packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -429,7 +547,7 @@ abstract class ResidentRunner {
         printStatus('To simulate different operating systems, (defaultTargetPlatform), press "o".');
       }
     }
-    if (device.supportsScreenshot)
+    if (runningDevices.first.device.supportsScreenshot)
```

Comment:
Seems like what you should use here is "any", not "first", given the way the key is implemented.